src/lib/datepicker/datepicker.ts

```diff
@@ -380,6 +380,10 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
 
   /** Open the calendar as a dialog. */
   private _openAsDialog(): void {
+    if (this._dialogRef) {
```

Comment:
I think that when you're holding the key down, eventually some of them go through, because we reset some of the variables in a `setTimeout`. You can see in the gif how it can take a little bit of holding before the dialogs start stacking.

Reply:
This change feels like masking the problem rather than fixing it. What if we switch to doing it in `beforeClosed` instead?

Reply:
That one doesn't help much either. The only difference is `beforeClose` is fired when the animation starts (which is still async) and the other one fires when the animation is done. IMO doing this is fine since it's pretty low maintenance on our end and it handles an edge case.

Reply:
Ok, as long as we've tried other solutions. Can you add a comment explaining why this is necessary even though it looks like it would be handled above?

Reply:
Done.